File: react-spring/code/jvm/modules/domain/src/main/kotlin/pt/isel/daw/dicepoker/domain/games/Game.kt
package pt.isel.daw.dicepoker.domain.games

data class Game(
    val id: Int,
    val name: String,
    val description: String,
    val rounds: Int,
    val status: String,
    val minPlayers: Int,
    val maxPlayers: Int,
    val timeout: Long,
    val createdAt: Long,
    val updatedAt: Long? = null,
    val gameStateType: String,
    val gameState: GameState? = null,
)

File: react-spring/code/js/dicePoker/domain/Game.ts

export type Game = {
    id: number;
    name: string;
    description: string;
    maxPlayers: number;
    minPlayers: number;
    rounds: number;
    status: string;
    timeout: number;
    gameStateType: string;
    gameState: GameState;
}

export type GameState = {
    currentPlayerId: number;
    currentRound: number;
    gameId: number;
    maxRounds: number;
    playerList: [PlayerSummary];
    playerHands: {number: Hand} //number is the player id
    playersExcluded: [number];
    ante: number;
    playersPaidAnte: {number: number;};
    pot: number;
    roundWinner: [PlayerSummary];
    winners: [PlayerSummary];
    turnTimer: number;
    turnStarted: number;
    rollsForTurn: number;
}


export type PlayerSummary = {
    userId: number;
    balance: number;
    currentHand?: Hand;
    hasPlayed: boolean;
    isHost: boolean;
}

export type Hand = {
    dices: [Dice];
    handRank: string;
    points: number;
}

export type Dice = {
    face: string;
    points: number;
}